src/client/cli/commands/archive.js

```diff
@@ -115,22 +131,30 @@ function writeDataObjectsToTarball (client: RestClient, tarball: Object, objectI
     if (batch.length >= OBJECT_BATCH_SIZE) {
       const ids = batch
       batch = []
-      batchPromises.push(fetchObjectBatch(client, tarball, ids))
+      batchPromises.push(fetchObjectBatch(client, tarball, ids, allowErrors))
     }
   }
 
-  batchPromises.push(fetchObjectBatch(client, tarball, batch))
+  batchPromises.push(fetchObjectBatch(client, tarball, batch, allowErrors))
   return Promise.all(batchPromises)
 }
 
-function fetchObjectBatch (client: RestClient, tarball: Object, objectIds: Array<string>): Promise<*> {
+function fetchObjectBatch (client: RestClient, tarball: Object, objectIds: Array<string>, allowErrors: boolean): Promise<*> {
   if (objectIds.length < 1) return Promise.resolve()
 
   return client.batchGetDataStream(objectIds, false)
     .then(stream => new Promise((resolve, reject) => {
       stream.on('data', dataResult => {
         const key = objectIds.shift()
-        if (dataResult == null || typeof dataResult !== 'object' || dataResult.data == null) return
+        if (dataResult == null || typeof dataResult !== 'object' || dataResult.data == null) {
+          const msg = (dataResult && dataResult.error) ? dataResult.error : 'Unknown error'
+          if (allowErrors) {
+            printlnErr(`Error fetching object for ${key}: ${msg}`)
+            return
```

Comment:
It will still resolve when the stream ends (in the `stream.on('end', ...)` below.  That will probably be immediately after the error, since concat will close the stream after the error.